src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -30,7 +30,7 @@ internal sealed class UserDefinitions
         private readonly Features _features;
 
         // Exposing it so hosts can filter out the intellisense suggestions
-        public static readonly ISet<DType> RestrictedTypes = new HashSet<DType> { DType.DateTimeNoTimeZone, DType.ObjNull,  DType.Decimal };
+        public static readonly ISet<DType> RestrictedTypes = new HashSet<DType> { DType.DateTimeNoTimeZone, DType.ObjNull, DType.Decimal, DType.Hyperlink };
```

Comment:
>DType.Hyperlink [](http://example.com/codeflow?start=138&length=15)

I don't know if we already have, but we should have tests that assert that we would fail if we try to define a UDF using those types.